File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
  let ans = [];

  const N = progresses.length;

  let days = []; // 남은 날들 계산해서 넣어줄 리스트
  for (let i = 0; i < N; i++) {
    let leftDays = Math.ceil((100 - progresses[i]) / speeds[i]);
    days.push(leftDays);
  }

  let curr = days.shift();
  let cnt = 1;
  while (days.length > 0) {
    let tmp = days.shift();
    if (curr < tmp) {
      // 현재 기준보다 더 크면, 지금까지 cnt 넣어주고 초기화, curr 갱신
      ans.push(cnt);
      cnt = 1;
      curr = tmp;
    } else {
      // cnt++
      cnt++;
    }
  }

  if (cnt > 0) {
    ans.push(cnt);
  }

  return ans;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    N = len(progresses)
    ans = []

    arr = deque() # 언제 배포가 가능한지
    for i in range(N):
        arr.append(math.ceil((100 - progresses[i]) / speeds[i]))

    curr = arr.popleft()
    cnt = 1
    while arr:
        tmp = arr.popleft()
        if curr < tmp: # 배포 해주고, 기준 갱신해준다
            ans.append(cnt)
            cnt = 1
            curr = tmp
        else: # cnt 더해주고, tmp 기준 바꾼다
            cnt += 1
    
    ans.append(cnt)
    return ans